tests/test-classnames.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * WP Component Library Tests: Test_Classnames class
+ *
+ * @package WP_Component_Library
+ * @subpackage Tests
+ */
+
+use WP_Component_Library\Classnames;
+
+/**
+ * A class to test the behavior of the Classnames class.
+ *
+ * @package WP_Component_Library
+ * @subpackage Tests
+ */
+class Test_Classnames extends WP_UnitTestCase {
+	/**
+	 * A data provider for the test_get_classlist function. Returns an array of
+	 * arrays representing function arguments.
+	 *
+	 * This list of arguments was copied from the classnames test suite at
+	 * https://github.com/JedWatson/classnames/blob/master/tests to ensure
+	 * compatibility with the classnames npm package.
+	 *
+	 * @return array An array of arrays representing function arguments.
+	 */
+	public function data_get_classlist(): array {
+		return [
+			// It keeps object keys with truthy values.
```

Comment:
This is just a drive-by comment, and you might already be aware of it, but just in case, it's possible to name each set in a data provider by providing it with a string key, and that key will also be used in the output if the set fails, like `'It keeps object keys with truthy values' => [ ... ]`. That might provide a little extra bang for the buck of having already written these comments. Details here: https://phpunit.readthedocs.io/en/7.5/writing-tests-for-phpunit.html#data-providers.